### ISSUE TITLE
Bug: Javascript error: document.getElementById("google.visualiation.MotionChart-0") is null

### ISSUE THREAD

**Issue by orwant** [google/google-visualization-api-issues#130]
Original [issue 132](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132) created by orwant on 2009-11-19T12:21:40.000Z:

<b>What steps will reproduce the problem? Please provide a link to a</b>
<b>demonstration page if at all possible, or attach code.</b>

Go to a page that contains a motion chart. For example:
http://code.google.com/apis/visualization/documentation/gallery/motionchart.html.
Wait until the motion chart is loaded and click anywhere in the document.
In Firefox 3.0 / 3.5 the following error occurs:
document.getElementById(&quot;google.visualiation.MotionChart-0&quot;) is null

In Chrome 3 this error occurs:
Uncaught TypeError: Cannot call method 'onMouseUpOutsideSwf' of null

Note that the motion chart is working properly as far as I know. Only this
error occurs every time you click

<b>What component is this issue related to (PieChart, LineChart, DataTable,</b>
<b>Query, etc)?</b>
Motion chart

<b>Are you using the test environment (version 1.1)?</b>
<b>(If you are not sure, answer NO)</b>

NO

<b>What operating system and browser are you using?</b>
Firefox 3.0.14 / 3.5.5
Chrome 3
 
Windows XP SP 3

<b>*********************************************************</b>
<b>For developers viewing this issue: please click the 'star' icon to be</b>
<b>notified of future changes, and to let us know how many of you are</b>
<b>interested in seeing it resolved.</b>
<b>*********************************************************</b>

**Comment by orwant** [comment]
Comment [#1](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c1) originally posted by orwant on 2009-11-22T16:52:59.000Z:

I just submitted the same issue.

It's related to a typo in
http://www.google.com/uds/api/visualization/1.0/98f02d1a8a7afee38bbae4203d873235/default,motionchart+fr.I.js

(missing 'z' : visualiation instead of Visualization)

**Comment by orwant** [comment]
Comment [#2](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c2) originally posted by orwant on 2010-01-29T20:57:29.000Z:

I have this problem too and i'm not sure if it's related but i'm unable to start the 
motion chart with a predefined state instead of the default one (i should work more 
on finding why) anyway the problem doesn't look like it's just the typo because after 
i create the new motion chart like this:
var mchart = new 
google.visualization.MotionChart(document.getElementById('chart_div'));

my div contents changes to:

<div style="position: relative;" id="chart_div"><embed type="application/x-shockwave-
flash" src="http://www.google.com/uds/modules/gviz/1.0/motionchart/en/tlz-gviz.swf" 
id="google.visualiation.MotionChart-0_swf" name="google.visualiation.MotionChart-
0_swf" bgcolor="#FFFFFF" quality="high" scale="default" allowscriptaccess="always" 
wmode="opaque" flashvars="state=&amp;chartId=google.visualiation.MotionChart-
0&amp;js=blocking" width="1200" height="900"></div>


so the chart id is "google.visualiation.MotionChart-0_swf" without the "z" but the 
problem is the flashvar chartId is "google.visualiation.MotionChart-0" without the 
"z" too but without the "_swf" at the end too!!!

**Comment by orwant** [comment]
Comment [#3](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c3) originally posted by orwant on 2010-03-06T14:21:51.000Z:

With this lines errors in Firefox gone. Not a right solution, but from client side is
the only one I found.

var swf = document.getElementById("google.visualiation.MotionChart-0_swf");
if (swf) swf.id = 'google.visualiation.MotionChart-0';

**Comment by orwant** [comment]
Comment [#4](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c4) originally posted by orwant on 2010-05-10T07:24:50.000Z:

Still seeing this error in both Firefox 3.6 and Chrome 4

**Comment by orwant** [comment]
Comment [#5](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c5) originally posted by orwant on 2010-06-07T04:49:03.000Z:

Seeing this error in Firefox and Chrome as well.

**Comment by orwant** [comment]
Comment [#6](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c6) originally posted by orwant on 2010-06-12T20:47:25.000Z:

I have the same problem, and I'm using version 1.1.0, My biggest problem is that i have some Anchors objects and when i open a new window with the same chart those anchors doesnt work at all.

**Comment by orwant** [comment]
Comment [#7](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c7) originally posted by orwant on 2010-07-25T19:03:52.000Z:

Can we get a comment from a dev when this going to be corrected?

**Comment by orwant** [comment]
Comment [#8](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c8) originally posted by orwant on 2010-07-27T09:24:56.000Z:

Confirm having the same issue.

**Comment by orwant** [comment]
Comment [#9](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c9) originally posted by orwant on 2010-09-03T03:47:02.000Z:

I faced the same issue. Did you guys figure out any solution???

**Comment by orwant** [comment]
Comment [#10](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c10) originally posted by orwant on 2010-09-07T07:12:59.000Z:

I believe this was already fixed. Do you still see it happens now?

**Comment by orwant** [comment]
Comment [#11](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c11) originally posted by orwant on 2010-09-08T06:46:54.000Z:

Still getting an error in both browsers. Only in Firefox it says: "document.getElementById("google.visualization.MotionChart-0") is null". The only difference is that there is an extra 'z' in the message. 

Now using Firefox 3.6.8 and Chrome 6.0.472.55

**Comment by orwant** [comment]
Comment [#12](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c12) originally posted by orwant on 2010-09-14T22:28:15.000Z:

I'm experiencing the same problem in Firefox 3.6.3: 
"document.getElementById("google.visualization.MotionChart-0") is null"

Please reopen the issue.

**Comment by orwant** [comment]
Comment [#13](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c13) originally posted by orwant on 2010-09-15T07:41:36.000Z:

Where do you see this problem occurs, in what web page?
As far as we saw, this does not happen any more on both Firefox and Chrome.

**Comment by orwant** [comment]
Comment [#14](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c14) originally posted by orwant on 2010-09-15T08:46:23.000Z:

For example, it still happens in the visualization gallery: http://code.google.com/apis/visualization/documentation/gallery/motionchart.html

**Comment by orwant** [comment]
Comment [#15](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c15) originally posted by orwant on 2010-09-15T08:58:26.000Z:

See attachment.

**Comment by orwant** [comment]
Comment [#16](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c16) originally posted by orwant on 2010-12-09T15:23:17.000Z:

I've attached to my post a simple html page where I still see the error:

- Firefox 3.6.12: document.getElementById("google.visualization.MotionChart-0") is null
- Chrome 8.0.552.215: Uncaught TypeError: Cannot call method 'onMouseUpOutsideSwf' of null

I've tested the same page on three differents PC and browsers and the error appear on my dev machine only, the one with Flex Builder and a flash player debug version (installed months ago and updated from time to time). The GoogleMotion works BTW, but the javascript console is filled of errors: after an inner event is generated (like a click inside the animation) every time the component container is redrawn the error pop up.

Other info:
Windows-7 Home Premium (64bit)
Firefox plugin: Shockwave Flash 10.1.82.76
Java Firefox plugin: 6.0.220.4

Hope it helps to find out that nasty bug.

**Comment by orwant** [comment]
Comment [#17](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c17) originally posted by orwant on 2010-12-27T16:15:35.000Z:

Wow! a long thread of issue. No comment on resolving it.

Same issue @ my end as well.

FF 3.5.16
Error: document.getElementById("google.visualization.MotionChart-0") is null
Event: On mouse (click or up or down)

**Comment by orwant** [comment]
Comment [#18](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=132#c18) originally posted by orwant on 2011-01-03T17:34:04.000Z:

I want to put my two cents into this issue. I've stumbled upon it by noticing a component of mine which uses motion charts' statechange event, currently working fine in production and not working on my development box. After pondering a lot about what I could have changed (because I actually didn't in this devel cycle) I've tested a bit more.

It turns out that the error message appears always, even on Google's own servers. However, what's different is that getState works on production servers, while it doesn't on development. I think you notice a pattern here, so I copied the file uploaded by cappeli.mauro to a development server and to an Apache-driven integration server. As you probably have guessed, integration works, development doesn't.

However, I've found the solution. My production/integration/testing servers are all running via HTTPS, while development is running HTTP. I tried changing the JS-API URL from "https://www.google.com/jsapi" to "http://www.google.com/jsapi" (without 's') and it worked! I guess there is some kind of access restriction to HTTP contents for HTTPS scripts, but I guess Google knows those restrictions better than me :)

I'd like to ask people bitten by this to check whether they have HTTPS deployments and HTTPS JS-API URLs. Maybe it's a hint at why problems exist -- however, the actual error message doesn't go away and is probably a separate issue.
